custom_components/tahoma/coordinator.py

```diff
@@ -43,20 +45,26 @@ def __init__(
         self.listener_id = listener_id
 
     async def _async_update_data(self) -> Dict[str, Device]:
-        """Fetch data from Tahoma."""
+        """Fetch TaHoma data via event listener."""
         try:
             events = await self.client.fetch_event_listener(self.listener_id)
-        except Exception as exception:
-            raise UpdateFailed(f"Error communicating with the TaHoma API: {exception}")
-        else:
-            for event in events:
-                if event.name == "DeviceStateChangedEvent":
-                    for state in event.device_states:
-                        device = self.devices[event.deviceurl]
-                        if state.name not in device.states:
-                            device.states[state.name] = state
-                        device.states[state.name].value = self._get_state(state)
+        except (ServerDisconnectedError, NotAuthenticatedException):
+            await self.client.login()
+            self.listener_id = await self.client.register_event_listener()
+            self.devices = await self.client.get_devices(refresh=True)
```

Comment:
devices here is a dict: deviceurl - > device.

Reply:
@tetienne what I don't understand is that `self.devices` is a variable that already contains the initial response from `await self.client.get_devices()` in __init__. Thus, this should not be different?

Reply:
Nope, have a look here: https://github.com/iMicknl/ha-tahoma/blob/bb28a2bc3d89f7b8ed2c2746a21964450d5fe48f/custom_components/tahoma/coordinator.py#L44